src/main/java/uk/yermak/audiobookconverter/fx/FilesController.java

```diff
@@ -150,6 +150,17 @@ public void initialize() {
         filesChapters.getTabs().remove(chaptersTab);
 
         bookStructure.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+        bookStructure.getSelectionModel().getSelectedItems().addListener((ListChangeListener<TreeItem<Organisable>>) c -> {
+            List<MediaInfo> list = ConverterApplication.getContext().getSelectedMedia();
+            list.clear();
+            for (TreeItem<Organisable> item : c.getList()) {
+                if (item.getValue() instanceof Chapter chapter) {
+                    list.addAll(chapter.getMedia());
+                } else if (item.getValue() instanceof MediaInfoProxy mediaInfoProxy) {
```

Comment:
MediaInfo Interface should be used rather then Proxy which should never been accessed directly.

Reply:
Fixed